[CCR] Add RemoteClusterEdit route.

cjcenizal opened the pull request:
I also migrated validation functionality into RemoteClusterForm and extracted it into its own component.

Use the "Edit" button in the detail panel to edit the remote cluster.

![image](https://user-images.githubusercontent.com/1238659/48650741-4ff67a00-e9ac-11e8-9661-91ff09ac367f.png)

elasticmachine commented:
Pinging @elastic/kibana-management

jen-huang commented on x-pack/plugins/remote_clusters/public/sections/remote_cluster_form/remote_cluster_form.js:
```diff
@@ -308,17 +299,105 @@ export class RemoteClusterFormUi extends Component {
     );
   }
 
+  renderActions() {
```
it would be nice to render a cancel button if editing

jen-huang commented on x-pack/plugins/remote_clusters/public/sections/remote_cluster_list/detail_panel/detail_panel.js:
```diff
@@ -186,6 +191,32 @@ export class DetailPanelUi extends Component {
+        <EuiFlyoutFooter>
+          <EuiFlexGroup justifyContent="spaceBetween">
+            <EuiFlexItem grow={false}>
+              <EuiButton color="danger">
```
could we consolidate the Disconnect and Edit actions under a single Manage button? this will be consistent with our pattern in Index Management and Rollup Jobs

cjcenizal commented:
Thanks @jen-huang I addressed your feedback. Could you please take another look?

![image](https://user-images.githubusercontent.com/1238659/48653228-90f48b80-e9b8-11e8-8b25-226679a2324b.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/420/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/436/)
